drivers/clock_control/clock_stm32_ll_common.c

```diff
@@ -368,6 +371,36 @@ static int stm32_clock_control_init(struct device *dev)
 	/* Disable other clocks */
 	LL_RCC_HSI_Disable();
 	LL_RCC_MSI_Disable();
+#elif CONFIG_CLOCK_STM32_PLL_SRC_PLL2
+	int hse_bypass = LL_UTILS_HSEBYPASS_OFF;
+	u32_t PLL2Frequency;
+	LL_UTILS_PLLInitTypeDef s_PLL2InitStruct;
+
+#ifdef CONFIG_CLOCK_STM32_HSE_BYPASS
+	hse_bypass = LL_UTILS_HSEBYPASS_ON;
+#endif /* CONFIG_CLOCK_STM32_HSE_BYPASS */
+
+	/* get PLL2 configuration */
+	config_pll2_init(&s_PLLInitStruct, &s_PLL2InitStruct);
+
+	PLL2Frequency =
+		(CONFIG_CLOCK_STM32_HSE_CLOCK / CONFIG_CLOCK_STM32_PLL2_PREDIV2)
+		* CONFIG_CLOCK_STM32_PLL2_MULTIPLIER;
+
+	/* switch the input to use PLL2 instead of HSE directly */
+	s_PLLInitStruct.Prediv |= RCC_CFGR2_PREDIV1SRC << 4U;
+
+	/* Configure PLL2 */
+	LL_RCC_PLL_ConfigDomain_PLL2(
+		s_PLL2InitStruct.Prediv,
+		s_PLL2InitStruct.PLLMul);
+
+	/* Switch to PLL, via PLL2 with HSE as clock source */
+	LL_PLL_ConfigSystemClock_HSE(
+		PLL2Frequency,
+		hse_bypass,
+		&s_PLLInitStruct,
+		&s_ClkInitStruct);
```

Comment:
Don't we miss APB/AHB configuration at this point ?

Reply:
[`UTILS_EnablePLLAndSwitchSystem()`](https://github.com/zephyrproject-rtos/hal_stm32/blob/fc5cb1a/stm32cube/stm32f1xx/drivers/src/stm32f1xx_ll_utils.c#L517) seems to set [AHB](https://github.com/zephyrproject-rtos/hal_stm32/blob/fc5cb1a/stm32cube/stm32f1xx/drivers/src/stm32f1xx_ll_utils.c#L564), [APB1](https://github.com/zephyrproject-rtos/hal_stm32/blob/fc5cb1a/stm32cube/stm32f1xx/drivers/src/stm32f1xx_ll_utils.c#L572) and [APB2](https://github.com/zephyrproject-rtos/hal_stm32/blob/fc5cb1a/stm32cube/stm32f1xx/drivers/src/stm32f1xx_ll_utils.c#L573) for us, and it is called from inside `LL_PLL_ConfigSystemClock_HSE()` (i.e: here), so I think we're okay...

Certainly the PLL code from before this patch didn't deal with APB / AHB config.

Only the non-PLL sources seem to set the prescalers themselves for some reason... (i.e: `CONFIG_CLOCK_STM32_SYSCLK_SRC_HSE` / `CONFIG_CLOCK_STM32_SYSCLK_SRC_MSI` / `CONFIG_CLOCK_STM32_SYSCLK_SRC_HSI`)

Reply:
Right, I got confused with non PLL feed sysclock configs.
This is fine, my bad.

Reply:
I'm reopening the point as you can't use `LL_PLL_ConfigSystemClock_HSE()` in your case.
So you don't have access to `UTILS_EnablePLLAndSwitchSystem()`